scipy/signal/tests/test_spectral.py

```diff
@@ -430,11 +430,291 @@ def test_complex_32(self):
+
+    def test_pad_shorter_y(self):
+        x = np.zeros(12)
+        y = np.zeros(8)
+
+        f = np.linspace(0, 0.5, 7)
+        c = np.zeros(7,dtype=np.complex128)
+        f1, c1 = csd(x, y, nperseg=12)
+
+        assert_allclose(f, f1)
+        assert_allclose(c, c1)
+
+    def test_real_onesided_even(self):
+        x = np.zeros(16)
+        x[0] = 1
+        x[8] = 1
+        f, p = csd(x, x, nperseg=8)
+        assert_allclose(f, np.linspace(0, 0.5, 5))
+        assert_allclose(p, np.array([0.08333333, 0.15277778, 0.22222222,
+            0.22222222, 0.11111111]))
+
+    def test_real_onesided_odd(self):
+        x = np.zeros(16)
+        x[0] = 1
+        x[8] = 1
+        f, p = csd(x, x, nperseg=9)
+        assert_allclose(f, np.arange(5.0)/9.0)
+        assert_allclose(p, np.array([0.15958226, 0.24193954, 0.24145223,
+            0.24100919, 0.12188675]))
+
+    def test_real_twosided(self):
+        x = np.zeros(16)
+        x[0] = 1
+        x[8] = 1
+        f, p = csd(x, x, nperseg=8, return_onesided=False)
+        assert_allclose(f, fftpack.fftfreq(8, 1.0))
+        assert_allclose(p, np.array([0.08333333, 0.07638889, 0.11111111,
+            0.11111111, 0.11111111, 0.11111111, 0.11111111, 0.07638889]))
+
+    def test_real_spectrum(self):
+        x = np.zeros(16)
+        x[0] = 1
+        x[8] = 1
+        f, p = csd(x, x, nperseg=8, scaling='spectrum')
+        assert_allclose(f, np.linspace(0, 0.5, 5))
+        assert_allclose(p, np.array([0.015625, 0.028645833333333332,
+            0.041666666666666664, 0.041666666666666664, 0.020833333333333332]))
+
+    def test_integer_onesided_even(self):
+        x = np.zeros(16, dtype=np.int)
+        x[0] = 1
+        x[8] = 1
+        f, p = csd(x, x, nperseg=8)
+        assert_allclose(f, np.linspace(0, 0.5, 5))
+        assert_allclose(p, np.array([0.08333333, 0.15277778, 0.22222222,
+            0.22222222, 0.11111111]))
+
+    def test_integer_onesided_odd(self):
+        x = np.zeros(16, dtype=np.int)
+        x[0] = 1
+        x[8] = 1
+        f, p = csd(x, x, nperseg=9)
+        assert_allclose(f, np.arange(5.0)/9.0)
+        assert_allclose(p, np.array([0.15958226, 0.24193954, 0.24145223,
+            0.24100919, 0.12188675]))
+
+    def test_integer_twosided(self):
+        x = np.zeros(16, dtype=np.int)
+        x[0] = 1
+        x[8] = 1
+        f, p = csd(x, x, nperseg=8, return_onesided=False)
+        assert_allclose(f, fftpack.fftfreq(8, 1.0))
+        assert_allclose(p, np.array([0.08333333, 0.07638889, 0.11111111,
+            0.11111111, 0.11111111, 0.11111111, 0.11111111, 0.07638889]))
+
+    def test_complex(self):
+        x = np.zeros(16, np.complex128)
+        x[0] = 1.0 + 2.0j
+        x[8] = 1.0 + 2.0j
+        f, p = csd(x, x, nperseg=8)
+        assert_allclose(f, fftpack.fftfreq(8, 1.0))
+        assert_allclose(p, np.array([0.41666667, 0.38194444, 0.55555556,
+            0.55555556, 0.55555556, 0.55555556, 0.55555556, 0.38194444]))
```

Comment:
and similarly, in several place:
    
    f, p = csd(x, ...
               detrend=...